### PR TITLE
feat(falsy arbitrary): add arbitrary, tests, and docs

### PR DESCRIPTION
**Pull request by larrybotha**
## Why is this PR for?

Adding a `falsy` arbitrary as per #624 

## In a nutshell

✔️ New feature

## Potential impacts

NA

### CODE REVIEW

File: test/unit/check/arbitrary/FalsyArbitrary.spec.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            const arb = falsy();
+            for (let id = 0; id !== 10000; ++id) {
+              const g = arb.generate(mrng).value;
+              if (g === falsyValue || (isLiterallyNaN(g) && isLiterallyNaN(falsyValue))) return true;
+            }
+            return false;
+          }
+        )
+      ));
+  });
```
[the remark this answers is not in the document]
Comment (larrybotha) [reply test/unit/check/arbitrary/FalsyArbitrary.spec.ts:38]: ok cool, will remove this test

File: src/check/arbitrary/FalsyArbitrary.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { Arbitrary } from './definition/Arbitrary';
```
[the remark this answers is not in the document]
Comment (larrybotha) [reply src/check/arbitrary/FalsyArbitrary.ts:1]: doh! Didn't even think about the export. Will get to it this evening.

File: src/check/arbitrary/FalsyArbitrary.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import { constantFrom } from './ConstantArbitrary';
+
+type FalsyType = boolean | null | number | string | typeof NaN | undefined;
```
[the remark this answers is not in the document]
Comment (larrybotha) [reply src/check/arbitrary/FalsyArbitrary.ts:4]: ah ok cool, I didn't consider using the primitives themselves in the union, but that makes sense. Thanks for the suggestion. Will get to it this evening.

File: src/check/arbitrary/FalsyArbitrary.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+/**
+ * For falsy values:
+ * - ''
+ * - 0
+ * - NaN
+ * - false
+ * - null
+ * - undefined
+ */
+function falsy(): Arbitrary<FalsyType> {
```
[the remark this answers is not in the document]
Comment (larrybotha) [reply src/check/arbitrary/FalsyArbitrary.ts:15]: Alternatively a property with an array of enums - I'm not sure if other values should be considered, and the user is allowed to determine which values they actually want generated.

e.g.

```typescript
enum FalsyValue {
	bigint0 = 'big-int-0',
	//... other potential values?
}

fc.falsy({
	enabledValues: [FalsyValue.bigint0, ...]
})
```

**Comment by dubzzz** [comment]
Perfect 👌

**Comment by dubzzz** [comment]
I am going to log issues related to the remaining remarks related to bigint and constantFrom not properly typed

**Pull request merged**